contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -136,21 +137,22 @@ def _create_envelope(self, data_point, timestamp, properties):
         return envelope
 
     def shutdown(self):
-        # flush metrics on exit
-        self.export_metrics(stats_module.stats.get_metrics())
-        if self._atexit_handler is not None:
-            atexit.unregister(self._atexit_handler)
-            self._atexit_handler = None
+        # Flush the exporter thread
+        if self.exporter_thread:
+            self.exporter_thread.close()
+        # Shutsdown storage worker
+        self.storage.close()
 
 
 def new_metrics_exporter(**options):
     exporter = MetricsExporter(**options)
     producers = [stats_module.stats]
     if exporter.options.enable_standard_metrics:
         producers.append(standard_metrics.producer)
-    transport.get_exporter_thread(producers,
-                                  exporter,
-                                  interval=exporter.options.export_interval)
+    exporter.exporter_thread = transport.get_exporter_thread(
```

Comment:
add it to `__init__()`?

Reply:
We expect users to initialize via `new_metric_exporter` so it has to be in this function.